### PR TITLE
feat: tdd round pages + refactor to context api

### LINKED ISSUE #105: task: take a TDD pass on Round pages
**Details**
Review following components and add in TDD where applicable:
* creating a grant round
* fetching a list of grant rounds
* view specific round + view applications

### PR DESCRIPTION
**Pull request by chibie**
##### Description

Review and test following components:
- [x] creating a grant round
- [x] fetching a list of grant rounds
- [x] view specific round
- [ ] view applications

##### Refers/Fixes

closes #105

### CODE REVIEW

**Comment by vacekj** [comment]
Wrapping up this PR , `ViewApplicationPage` and `RoundApplicationForm` warrant another ticket with at least 2 SPs - will put that in the backlog.

**Comment by vacekj** [comment]
@thelostone-mc done

**Review by thelostone-mc — APPROVED** [review]

File: packages/round-manager/src/context/RoundContext.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  const { address, provider } = useWallet();
+
+  useEffect(() => {
+    if (roundId) {
+      fetchRoundById(context.dispatch, provider, roundId);
+    }
+  }, [address, provider, roundId, context.dispatch]);
+
+  return {
+    round: context.state.data.find((round) => round.id === roundId),
```
Comment (dschinkel) [inline packages/round-manager/src/context/RoundContext.tsx:128]: can this just be context.state.data[0]?

**Review by shavinac — APPROVED** [review]
FYI I rebased & resolved a minor merge conflict with `main` for the `CopyToClipboard` button

Code changes LGTM but the commit message seems wrong? Could you reword the commit message to refer to the story?

File: packages/round-manager/src/context/RoundContext.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  const context = useContext(RoundContext);
+  if (context === undefined) {
+    throw new Error("useRounds must be used within a RoundProvider");
+  }
+  const { address, provider } = useWallet();
+
+  useEffect(() => {
+    if (roundId) {
+      const existingRound = context.state.data.find(
+        (round) => round.id === id
```
Comment (shavinac) [inline packages/round-manager/src/context/RoundContext.tsx:124]: my bad, used the wrong variable name in the previous suggestion 🙇‍♀️ 
```suggestion
        (round) => round.id === roundId
```

**Pull request merged**